snapcraft/internal/pluginhandler.py

```diff
@@ -656,15 +656,27 @@ def _migrate_files(snap_files, snap_dirs, srcdir, dstdir, missing_ok=False,
     for directory in snap_dirs:
         src = os.path.join(srcdir, directory)
         dst = os.path.join(dstdir, directory)
+
+        dir_stat = os.stat(src, follow_symlinks=follow_symlinks)
+        uid = dir_stat.st_uid
+        gid = dir_stat.st_gid
         os.makedirs(dst, exist_ok=True)
+        os.chown(dst, uid, gid, follow_symlinks=follow_symlinks)
         shutil.copystat(src, dst, follow_symlinks=follow_symlinks)
```

Comment:
Looking good. Though lines 660-665 look ripe for extraction into a function that can also be used on 671-679, no?